e2e-test/src/tests/analyze.test.js

```diff
@@ -0,0 +1,106 @@
+    selectedMetabolites = await page.$x("//div[@class='v-input v-input--selection-controls v-input--checkbox v-input--hide-details v-input--is-label-active v-input--is-dirty theme--light']");
+    await expect(selectedMetabolites).toHaveLength(1);
+
+    await expect(page).toClickXPath("//*[text()='C26 - Fol']" + vBtn({ title: 'Adds the highlighted Metabolites to the selected set' }));
+    selectedMetabolites = await page.$x("//div[@class='v-input v-input--selection-controls v-input--checkbox v-input--hide-details v-input--is-label-active v-input--is-dirty theme--light']");
+    await expect(selectedMetabolites).toHaveLength(4);
+
+    await expect(page).toClickXPath("//*[text()='C26 - Veh']" + vBtn({ title: 'Adds the highlighted Metabolites to the selected set' }));
+    selectedMetabolites = await page.$x("//div[@class='v-input v-input--selection-controls v-input--checkbox v-input--hide-details v-input--is-label-active v-input--is-dirty theme--light']");
+    await expect(selectedMetabolites).toHaveLength(7);
+
+    await expect(page).toClickXPath("//*[text()='C26.Fol - Fol']" + vBtn({ title: 'Adds the highlighted Metabolites to the selected set' }));
+    selectedMetabolites = await page.$x("//div[@class='v-input v-input--selection-controls v-input--checkbox v-input--hide-details v-input--is-label-active v-input--is-dirty theme--light']");
+    await expect(selectedMetabolites).toHaveLength(8);
+
+    await expect(page).toClickXPath("//*[text()='C26.Fol - Veh']" + vBtn({ title: 'Adds the highlighted Metabolites to the selected set' }));
+    selectedMetabolites = await page.$x("//div[@class='v-input v-input--selection-controls v-input--checkbox v-input--hide-details v-input--is-label-active v-input--is-dirty theme--light']");
+    await expect(selectedMetabolites).toHaveLength(11);
+
+    await expect(page).toClickXPath("//*[text()='Fol - Veh']" + vBtn({ title: 'Adds the highlighted Metabolites to the selected set' }));
+    selectedMetabolites = await page.$x("//div[@class='v-input v-input--selection-controls v-input--checkbox v-input--hide-details v-input--is-label-active v-input--is-dirty theme--light']");
+    await expect(selectedMetabolites).toHaveLength(12);
+  });
+  it('select and unselect filters', async () => {
+    await expect(page).toClickXPath(vBtn('Your Datasets'));
+
+    await uploadDataset('tableTest');
+    await expect(page).toClickXPath(vListTile({ title: 'tableTest.csv ', content: 'Dataset ready for analysis.' }) + vBtn('View Data'));
+
+    // click download data
+    await expect(page).toClickXPath(vListTile({ title: 'Download Data' }));
+
+    // unselect 'Selected' for Metabolite Filter
+    await expect(page).toClickXPath("//input[@aria-checked='true'][@aria-label='Selected (0)']");
+
+    // verify unselected
+    await page.waitForXPath("//input[@aria-checked='false'][@aria-label='Selected (0)']");
```

Comment:
These verifies are just checking that the select button is rendered properly as unchecked, not that the table is actually deselecting anything.
There's an element in the top left of the table: `//div[@class='column-header']/div[@class='column-header-cell']`
It shows the dimensions of the table, like `26 x 31`. Every time I select/deselect something, the dimensions change. You should be able to use that as thing to assert on.